doc/source/examples/sharp-immersed-boundary-solver/3d-opencascade-mixer-with-pbt-impeller/3-opencascade-mixer-with-pbt-impeller.rst

```diff
@@ -0,0 +1,134 @@
+
+
+Definition of the shape and its motion
+--------------------------------------
+
+The section defining each parameter for the particles has certains requirements:
+
+1. ``length ratio`` defines the length used to apply the immersed boundaries through interpolation. It should stay as low as possible, but above ``1``.
+2. ``type`` and ``shape arguments`` are used to declare that the shape is an ``opencascade`` and that its data is located in ``impeller.step``.
+3. ``integrate motion`` is set to ``false``. This way, the solid only moves according to the prescribed `orientation` and angular velocity `omega` (the alternative being the integration of particle movement from forces).
+
+.. code-block:: text
+
+    subsection particles
+      set number of particles                           = 1
+      set stencil order                                 = 2
+      set refine mesh inside radius factor              = 0.0
+      set refine mesh outside radius factor             = 1.1
+      set length ratio                                  = 3
+      set initial refinement                            = 4
+      set integrate motion                              = false
+      set assemble Navier-Stokes inside particles       = false
+      set enable extra sharp interface vtu output field = true
+
+      subsection particle info 0
+        subsection position
+          set Function expression = -0.25;0;0
+        end
+        subsection velocity
+          set Function expression = 0;0;0
+        end
+        subsection orientation
+          set Function expression = -1*2*pi*t;pi/2;0
+        end
+        subsection omega
+          set Function expression = -1*2*pi;0;0
+        end
+        set type            = opencascade
+        set shape arguments = impeller.step
+      end
+    end
+
+Additionnal information on the ``particles`` parameters can be found on :doc:`../../../parameters/sharp-immersed-boundary-solver/sharp-immersed-boundary-solver`.
+
+Boundary conditions
+-----------------------
+
+Because the particles defined for the sharp solver are not divergence-free, it is necessary to have at least one boundary condition that is weakly imposed to ensure the system of equations is well-posed. For this purpose, a ``function weak`` type of boundary is used.
+Two aspects need special consideration:
+
+1. ``function weak`` is a variation of ``function``. It is used to weakly imposed a Dirichlet boundary condition, and it is necessary when using ``gls_sharp_navier_stokes_3d``.
+2. ``beta`` has to be defined. It is a Nitsche penalization parameter that enforces more strongly the boundary condition when it increases (see :doc:`../../../parameters/cfd/nitsche`).
```

Comment:
I have a feeling that this repeats a bit the other example. Why not refer to the other example in the top?

Reply:
You're right, I'll change it